content/_code-samples/send-a-memo/js/send-a-memo.js

```diff
@@ -0,0 +1,53 @@
+if (typeof module !== "undefined") {
+    // Use var here because const/let are block-scoped to the if statement.
+    var xrpl = require('xrpl')
+  }
+  // Validate and send a Memo
+  // https://xrpl.org/transaction-common-fields.html#memos-field
+  
+  async function main() {
+    // Connect to a testnet node
+    console.log("Connecting to Testnet...")
+    const client = new xrpl.Client('wss://s.altnet.rippletest.net:51233')
+    await client.connect()
+  
+    // Get account credentials from the Testnet Faucet
+    console.log("Requesting addresses from the Testnet faucet...")
+    const { wallet, balance } = await client.fundWallet()
+    
+    console.log("Account: ", wallet.address)
+    console.log("   Seed: ", wallet.seed)
+
+    // Enter memo data to insert into a transaction
+    const MemoData = Buffer.from("TEXT 123 -+=", 'utf8').toString('hex')
+    const MemoType = Buffer.from("Text", 'utf8').toString('hex')
+    const MemoFormat = Buffer.from("text/csv", 'utf8').toString('hex')
+
+    // Send AccountSet transaction 
+    const prepared = await client.autofill({
+      "TransactionType": "Payment",
+      "Account": wallet.address,
+      "Destination": "rPT1Sjq2YGrBMTttX4GZHjKu9dyfzbpAYe",
+      "Amount": "1000000", // Amount in drops, 1 XRP = 1,000,000 drops
+      "Memos": [{
+            "Memo":{
+                "MemoType": MemoType,
+                "MemoData": MemoData,
+                "MemoFormat": MemoFormat
+            }
+        }]
+    })
+
+    const signed = wallet.sign(prepared)  
+    const submit_result = await client.submitAndWait(signed.tx_blob)
+
+    console.log(`\n Transaction MEMO: ${JSON.stringify({"MemoType": MemoType, "MemoData": MemoData, "MemoFormat": MemoFormat})}`)
+    console.log("\n Transaction hash:", signed.hash)
+    console.log("    Submit result:", submit_result.result.meta.TransactionResult)
```

Comment:
Can you add a log that shows that the memo was encoded properly by showing the memo decoded derived from `submit_result`?